### PR TITLE
Generate pcre methodsynopses based on stubs

### PR DESCRIPTION
**Pull request by kocsismate**

### CODE REVIEW

File: reference/pcre/functions/preg-filter.xml [hunk 1]
```diff
@@ -9,12 +9,12 @@
...
-   <methodparam><type>mixed</type><parameter>pattern</parameter></methodparam>
-   <methodparam><type>mixed</type><parameter>replacement</parameter></methodparam>
-   <methodparam><type>mixed</type><parameter>subject</parameter></methodparam>
+   <type class="union"><type>string</type><type>array</type><type>null</type></type><methodname>preg_filter</methodname>
+   <methodparam><type class="union"><type>string</type><type>array</type></type><parameter>pattern</parameter></methodparam>
+   <methodparam><type class="union"><type>string</type><type>array</type></type><parameter>replacement</parameter></methodparam>
+   <methodparam><type class="union"><type>string</type><type>array</type></type><parameter>subject</parameter></methodparam>
    <methodparam choice="opt"><type>int</type><parameter>limit</parameter><initializer>-1</initializer></methodparam>
-   <methodparam choice="opt"><type>int</type><parameter role="reference">count</parameter></methodparam>
+   <methodparam choice="opt"><type>int</type><parameter role="reference">count</parameter><initializer>&null;</initializer></methodparam>
```
[the remark this answers is not in the document]
Comment (kocsismate) [reply reference/pcre/functions/preg-filter.xml:17]: I think so, because that's what did all the way down

[the remark this answers is not in the document]
Comment (cmb69) [reply reference/pcre/functions/preg-filter.xml:17]: My point is that you can't actually pass `null` as argument, e.g. https://3v4l.org/ojRU5. /cc @nikic

[the remark this answers is not in the document]
Comment (nikic) [reply reference/pcre/functions/preg-filter.xml:17]: You can't pass it, but this is still how you write an optional by-ref argument in PHP. I think we should follow in docs what you would write in code.

[the remark this answers is not in the document]
Comment (cmb69) [reply reference/pcre/functions/preg-filter.xml:17]: D'oh. Thanks!

File: reference/pcre/functions/preg-match-all.xml [hunk 2]
```diff
@@ -9,11 +9,11 @@
  <refsect1 role="description">
   &reftitle.description;
   <methodsynopsis>
-   <type>int</type><methodname>preg_match_all</methodname>
+   <type class="union"><type>int</type><type>false</type><type>null</type></type><methodname>preg_match_all</methodname>
    <methodparam><type>string</type><parameter>pattern</parameter></methodparam>
    <methodparam><type>string</type><parameter>subject</parameter></methodparam>
-   <methodparam choice="opt"><type>array</type><parameter role="reference">matches</parameter></methodparam>
-   <methodparam choice="opt"><type>int</type><parameter>flags</parameter><initializer><constant>PREG_PATTERN_ORDER</constant></initializer></methodparam>
+   <methodparam choice="opt"><type>array</type><parameter role="reference">matches</parameter><initializer>&null;</initializer></methodparam>
```
Comment (cmb69) [inline reference/pcre/functions/preg-match-all.xml:15]: This changes matches the stubs, and the stubs are not wrong, but it seems to me that `PREG_PATTERN_ORDER` is the more helpful default value.

**Comment by cmb69** [comment]
Thanks! Applied as http://svn.php.net/viewvc?view=revision&revision=351490.

**Pull request closed without merging**